packages/blade/scripts/generateBundleSizeInfo.js

```diff
@@ -0,0 +1,115 @@
+const execa = require('execa');
+
+const main = async () => {
+  const { globby } = await import('globby');
+
+  const indexPaths = await globby([
+    'build/lib/web/production/components/**/index.js',
+    '!build/lib/web/production/components/{Icons,Form}/**/index.js',
+    '!build/lib/web/production/components/index.js',
+    '!build/lib/web/production/components/Box/styledProps/index.js',
+  ]);
+
+  const excludedComponents = [
+    'useTheme',
+    'announce',
+    'clearAnnouncer',
+    'destroyAnnouncer',
+    'getTextProps',
+    'screenReaderStyles',
+    'useActionListContext',
+    'ActionListItemAsset', // This is just an <img> tag
+    // These are same as Badge, Amount, Counter, Link, Text components
+    'CardHeaderAmount',
+    'CardHeaderBadge',
+    'CardHeaderCounter',
+    'CardHeaderLink',
+    'CardHeaderText',
+    'ComponentIds',
+    // Accessibility components
+    'SkipNavContent',
+    'SkipNavLink',
+    'VisuallyHidden',
+  ];
+  const sizes = [];
+
+  // Get all the components name exported from the bundle and add them to the size-limit configuration
+  indexPaths.forEach((indexPath) => {
+    const fileContent = fs.readFileSync(path.resolve(__dirname, `../${indexPath}`), 'utf8');
+
+    // Parse the file content to generate an AST
+    const ast = babelParser.parse(fileContent, {
+      sourceType: 'module',
+    });
+    // Arrays to store size-limit stats and component names to exclude
+    const exportedComponents = [];
+
+    // Traverse the AST to get the component names
+    traverse(ast, {
+      // Get the component name from the export statement
+      ExportSpecifier: ({ node }) => {
+        const componentName = node.exported.name;
+
+        // We don't want to add Icon components to the size-limit configuration
+        if (
+          !(
+            excludedComponents.includes(componentName) ||
+            componentName.startsWith('Base') ||
+            componentName === 'default'
+          )
+        ) {
+          exportedComponents.push(componentName);
+        }
+      },
+    });
+
+    if (exportedComponents.length > 0) {
+      const imports = exportedComponents.join(', ');
+      // Write size-limit configuration to .size-limit.json for each component
+      fs.writeFileSync(
+        path.resolve(__dirname, '../.size-limit.json'),
+        JSON.stringify(
+          [
+            {
+              name: imports,
+              path: './build/lib/web/production/components/index.js',
+              import: `{ ${exportedComponents.join(', ')} }`,
+              // Set high limit for the component size so that it doesn't fail the size-limit check
+              limit: '2000 kb',
+              running: false,
+              gzip: true,
+            },
+          ],
+          null,
+          2,
+        ),
+      );
+
+      // Run size-limit command and capture the output to gather size information
+      const { stdout } = execa.commandSync('yarn size-limit --json');
```

Comment:
The sizes for each component that we recieve here is not the size of the component but the size of the project when using only this component. This would be a misleading stat to post as a comment on the PR. Lets do either
1. Measure the size of the component only (this would mean using something other than size-limit)
2. Measure the size of the component & the size of an empty size-limit project and use the diff as the component size
3. Mention a note in the comment saying that the sizes mentioned in the table are not actual component sizes but rather size of the project when using only this component

Reply:
@chaitanyadeorukhkar I've updated the PR with 2nd approach.